webpack/scenes/RedHatRepositories/components/EnabledRepository.js

```diff
@@ -1,42 +1,103 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
-import { ListView } from 'patternfly-react';
+import axios from 'axios';
+import { ListView, Spinner, OverlayTrigger, Tooltip } from 'patternfly-react';
 import { getTypeIcon } from '../../../services/index';
+import { connect } from 'react-redux';
 
-const EnabledRepository = ({
-  arch, name, id, type, releasever,
-}) => (
-  <ListView.Item
-    key={id}
-    actions={
-      // eslint-disable-next-line
-      <a
-        onClick={() => {
-          // https://github.com/Katello/katello/blob/c0d72eb79981c215a664021bf90ef79eb2a286d2/app/controllers/katello/api/v2/repository_sets_controller.rb#L64
-        }}
-      >
-        <i className={cx('fa-2x', 'fa fa-minus-circle')} />
-      </a>
-    }
-    leftContent={<ListView.Icon name={getTypeIcon(type)} />}
-    additionalInfo={[
-      <ListView.InfoItem key="1">
-        <strong>{type.toUpperCase()}</strong>
-      </ListView.InfoItem>,
-    ]}
-    heading={__(name)}
-    description={`${arch} ${releasever}`}
-    stacked
-  />
-);
+import { setRepositoryDisabled } from '../../../redux/actions/RedHatRepositories/enabled';
+
+class EnabledRepository extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = { loading: false };
+  }
+
+  setDisabled = response => {
+    const { productId, contentId, arch, releasever, name, type } = this.props;
+
+    const disabledRepo = {
+      contentId,
+      productId,
+      name,
+      type,
+      arch,
+      releasever,
+    };
+
+    this.props.setRepositoryDisabled(disabledRepo);
+  };
+
+  disableRepository = () => {
+    this.setState({ loading: true });
+
+    const { productId, contentId, arch, releasever } = this.props;
+
+    // https://github.com/Katello/katello/blob/c0d72eb79981c215a664021bf90ef79eb2a286d2/app/controllers/katello/api/v2/repository_sets_controller.rb#L64
+    const url = `/katello/api/v2/products/${productId}/repository_sets/${contentId}/disable`;
```

Comment:
At some point it would be good to think about how we can factor out the `/katello/api/v2/` part of this and have it in one place such that when the path changes we only have to change it in one place.